keras/backend/tensorflow_backend.py

```diff
@@ -2831,14 +2836,25 @@ def conv1d(x, kernel, strides=1, padding='valid',
         x: Tensor or variable.
         kernel: kernel tensor.
         strides: stride integer.
-        padding: string, `"same"` or `"valid"`.
+        padding: string, `"same"`, `"causal"` or `"valid"`.
         data_format: string, one of "channels_last", "channels_first".
         dilation_rate: integer dilate rate.
 
     # Returns
         A tensor, result of 1D convolution.
     """
     # pre-process dtype
+    if hasattr(kernel, '_keras_shape'):
```

Comment:
This is the TF backend, so don't rely on `_keras_shape`, instead use `tuple(kernel.get_shape().as_list())` (always available).